pkg/agent/ipassigner/linkdetector/link_detector.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linkdetector
+
+type LinkEventHandler func(linkName string)
+
+type Interface interface {
+	LinkExists(linkName string) bool
+
+	// Run starts the detector.
+	Run(stopCh <-chan struct{})
+
+	// AddEventHandler registers an eventHandler of link updates. It's not thread-safe and should be called before
+	// starting the detector.
+	AddEventHandler(handler LinkEventHandler, linkName ...string)
+
+	// HasSynced returns true if the cache has been initialized with the existing links.
+	HasSynced() bool
+}
```

Comment:
@tnqn, do you think this interface is appropriate to address the issue and https://github.com/antrea-io/antrea/issues/6547 as well? I did not cache the `netlink.Link` structs in the detector for simplicity.

Reply:
Hi @tnqn, could you comment on this? Thanks a lot!

Reply:
I haven't thought about how to leverage it in #6547 yet but it looks good to me for now.
Please design a minimum interface in a generic way that is best for addressing your current issue, I could add new methods or update it when using.